apps/browser/qml/pages/components/TabView.qml

```diff
@@ -40,131 +34,230 @@ SilicaGridView {
 
-        enabled: !closingAllTabs
-        opacity: enabled ? 1.0 : 0.0
-        Behavior on opacity { FadeAnimator {}}
+    anchors.fill: parent
 
-        width: thumbnailWidth
-        height: thumbnailHeight
+    Private.TabView {
+        id: tabs
 
-        GridView.onAdd: AddAnimation {
-            target: tabItem
+        anchors {
+            fill: parent
+            bottomMargin: tabsToolBar.height
         }
-        GridView.onRemove: RemoveAnimation {
-            target: tabItem
+
+        header: Private.TabBar {
+            model: headerModel
+            Rectangle {
+                anchors {
+                    fill: parent
+                    topMargin: _pulleyMenuApparentHeight
+                }
+
+                z: -100
+                color: Theme.colorScheme == Theme.LightOnDark ? "black" : "white"
+            }
         }
-    }
 
-    // Behind tab delegates
-    children: [
-        PrivateModeTexture {
-            z: -1
-            visible: opacity > 0.0
-            opacity: privateMode ? 1.0 : 0.0
-
-            Behavior on opacity { FadeAnimation {} }
-        },
-
-        MouseArea {
-            z: -1
-            width: tabView.width
-            height: tabView.height
-            onClicked: hide()
+        model: [ persistentTabView, privateTabView ]
+        interactive: false
+
+        onCurrentIndexChanged: {
+            if (_remorsePopup) {
+                _remorsePopup.trigger()
+            }
+            privateMode = currentIndex !== 0
+        }
+        Component.onCompleted: {
+            persistentIcon.updateGrubImage()
+            privateIcon.updateGrubImage()
+            currentIndex = privateMode ? 1 : 0
         }
-    ]
 
-    ConfigurationValue {
-        id: showCloseAllAction
-        key: "/apps/sailfish-browser/settings/show_close_all"
-        defaultValue: true
-    }
+        Component {
+            id: persistentTabView
 
-    PullDownMenu {
-        id: pullDownMenu
-
-        flickable: tabView
-
-        MenuItem {
-            text: tabView.privateMode ?
-                    //: Menu item switching back to normal browser
-                    //% "Normal browsing"
-                    qsTrId("sailfish_browser-me-normal_browsing") :
-                    //: Menu item switching to private browser
-                    //% "Private browsing"
-                    qsTrId("sailfish_browser-me-private_browsing")
-            onDelayedClick: {
-                if (remorsePopup) {
-                    remorsePopup.trigger()
+            Private.TabItem {
+                allowDeletion: false
+                flickable: _persistentTabView
+
+                TabGridView {
+                    id: _persistentTabView
+                    privateMode: false
+                    portrait: root.portrait
+                    model: webView.persistentTabModel
+                    header: Item {
+                        width: 1
+                        height: tabs.tabBarHeight
```

Comment:
In order to compensate for removing the y offset from `TabGridView.qml` I suggest adding the extra space here instead.
```suggestion
                        height: tabs.tabBarHeight + Theme.paddingLarge
```

Reply:
done